diff --git a/org.talend.datacleansing.core.ui/main/java/src/org/talend/datacleansing/core/ui/util/LibraryDeploymentHelper.java b/org.talend.datacleansing.core.ui/main/java/src/org/talend/datacleansing/core/ui/util/LibraryDeploymentHelper.java
index 6933720..37cb73c 100644
--- a/org.talend.datacleansing.core.ui/main/java/src/org/talend/datacleansing/core/ui/util/LibraryDeploymentHelper.java
+++ b/org.talend.datacleansing.core.ui/main/java/src/org/talend/datacleansing/core/ui/util/LibraryDeploymentHelper.java
@@ -14,8 +14,12 @@ import org.eclipse.emf.common.util.EList;
 import org.talend.commons.exception.ExceptionHandler;
 import org.talend.commons.utils.WorkspaceUtils;
 import org.talend.core.CorePlugin;
+import org.talend.core.classloader.ClassLoaderFactory;
 import org.talend.core.database.conn.version.EDatabaseVersion4Drivers;
 import org.talend.core.model.general.ILibrariesService;
+import org.talend.core.model.metadata.IMetadataConnection;
+import org.talend.core.model.metadata.builder.ConvertionHelper;
+import org.talend.core.model.metadata.builder.connection.Connection;
 import org.talend.core.model.metadata.builder.connection.DatabaseConnection;
 import org.talend.core.model.metadata.builder.connection.DelimitedFileConnection;
 import org.talend.core.model.metadata.builder.connection.MDMConnection;
@@ -124,6 +128,13 @@ public class LibraryDeploymentHelper {
                             ExceptionHandler.process(e);
                         }
                     }
+                } else if (SupportDBUrlType.HIVEDEFAULTURL.getDBKey().equals(dbConn.getDatabaseType())) {
+                    IMetadataConnection metadataConn = ConvertionHelper.convert((Connection) connection);
+                    String[] moduleList = ClassLoaderFactory.getDriverModuleList(metadataConn);
+
+                    for (String moduleName : moduleList) {
+                        jarNameSet.add(moduleName);
+                    }
                 } else { // other supported databases
                     List<String> driverJarNameList = EDatabaseVersion4Drivers.getDrivers(dbConn.getDatabaseType(),
                             dbConn.getDbVersionString());
diff --git a/org.talend.dataprofiler.core.tdq/src/org/talend/dataprofiler/core/tdq/action/LocalReportDocGenerator.java b/org.talend.dataprofiler.core.tdq/src/org/talend/dataprofiler/core/tdq/action/LocalReportDocGenerator.java
index 2872abe..18d2137 100644
--- a/org.talend.dataprofiler.core.tdq/src/org/talend/dataprofiler/core/tdq/action/LocalReportDocGenerator.java
+++ b/org.talend.dataprofiler.core.tdq/src/org/talend/dataprofiler/core/tdq/action/LocalReportDocGenerator.java
@@ -23,7 +23,9 @@ import org.talend.commons.exception.LoginException;
 import org.talend.commons.exception.PersistenceException;
 import org.talend.core.model.properties.Item;
 import org.talend.core.model.properties.Property;
+import org.talend.core.repository.model.IRepositoryFactory;
 import org.talend.core.repository.model.ProxyRepositoryFactory;
+import org.talend.core.repository.model.RepositoryFactoryProvider;
 import org.talend.dataprofiler.core.tdq.i18n.Messages;
 import org.talend.dataquality.analysis.Analysis;
 import org.talend.dataquality.properties.TDQReportItem;
@@ -36,6 +38,7 @@ import org.talend.dq.helper.resourcehelper.RepResourceFileHelper;
 import org.talend.dq.nodes.ReportRepNode;
 import org.talend.repository.ProjectManager;
 import org.talend.repository.RepositoryWorkUnit;
+import org.talend.repository.model.RepositoryConstants;
 import org.talend.utils.sugars.ReturnCode;
 
 /**
@@ -112,7 +115,14 @@ public class LocalReportDocGenerator extends ReportDocGenerator {
 
     @Override
     public ReturnCode generate() {
+        // ADD msjian TDQ-7707 2013-8-1: won't invoke any remote repository action such as svn commit(same to TDQ-7508)
+        IRepositoryFactory localRepository = RepositoryFactoryProvider
+                .getRepositoriyById(RepositoryConstants.REPOSITORY_LOCAL_ID);
+        IRepositoryFactory oldRepository = ProxyRepositoryFactory.getInstance().getRepositoryFactoryFromProvider();
+        ProxyRepositoryFactory.getInstance().setRepositoryFactoryFromProvider(localRepository);
         ReturnCode rc = super.generate();
+        ProxyRepositoryFactory.getInstance().setRepositoryFactoryFromProvider(oldRepository);
+        // TDQ-7707~
 
         // ADD msjian TDQ-4234: save the ExecutionInformations into report object
         Display.getDefault().asyncExec(new Runnable() {
diff --git a/org.talend.designer.components.tdqprovider/components/tDqReportRun/org.talend.core.runtime.jar b/org.talend.designer.components.tdqprovider/components/tDqReportRun/org.talend.core.runtime.jar
index fd8feb6..d810a53 100644
Binary files a/org.talend.designer.components.tdqprovider/components/tDqReportRun/org.talend.core.runtime.jar and b/org.talend.designer.components.tdqprovider/components/tDqReportRun/org.talend.core.runtime.jar differ
diff --git a/org.talend.designer.components.tdqprovider/components/tDqReportRun/org.talend.metadata.managment.jar b/org.talend.designer.components.tdqprovider/components/tDqReportRun/org.talend.metadata.managment.jar
index 58f3ca0..e33d467 100644
Binary files a/org.talend.designer.components.tdqprovider/components/tDqReportRun/org.talend.metadata.managment.jar and b/org.talend.designer.components.tdqprovider/components/tDqReportRun/org.talend.metadata.managment.jar differ
diff --git a/org.talend.designer.components.tdqprovider/components/tDqReportRun/tDqReportRun_java.xml b/org.talend.designer.components.tdqprovider/components/tDqReportRun/tDqReportRun_java.xml
index ac3a3ea..8dc189f 100644
--- a/org.talend.designer.components.tdqprovider/components/tDqReportRun/tDqReportRun_java.xml
+++ b/org.talend.designer.components.tdqprovider/components/tDqReportRun/tDqReportRun_java.xml
@@ -100,6 +100,7 @@
         <IMPORT NAME="mdm_webservice" MODULE="mdm_webservice.jar" REQUIRED="true" />
         <IMPORT NAME="Zql" MODULE="Zql.jar" UrlPath="platform:/plugin/org.talend.sqltools/lib/Zql.jar" REQUIRED="true" />
         <IMPORT NAME="javacsv" MODULE="javacsv.jar" UrlPath="platform:/plugin/org.talend.libraries.javacsv/lib/javacsv.jar" REQUIRED="true" />
+        <IMPORT NAME="serializer-2.7.1" MODULE="serializer-2.7.1.jar" UrlPath="platform:/base/plugins/org.apache.xml.serializer_2.7.1.v201005080400.jar" REQUIRED="true" />
         
       	<!-- Hibernate libraries -->
         <IMPORT NAME="hibernate-3.2-patched1.6" MODULE="hibernate-3.2-patched1.6.jar" UrlPath="platform:/plugin/org.talend.libraries.hibernate/lib/hibernate-3.2-patched1.6.jar" REQUIRED="true"/>
